user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md

```diff
@@ -0,0 +1,139 @@
+---
+
+# Securing NATS
+
+By default, NATS does **not** employ TLS encryption, leaving communication susceptible to eavesdropping. Consequently, we **strongly recommend enabling TLS encryption** for enhanced security within your NATS cluster.
+
+## Enabling NATS inter-node TLS communication
+
+> [!NOTE]
+> This applies solely to instances involving a DMS cluster or external DxMs. If you have a single-Agent setup, all communication is confined to the local host, as there is only one NATS node.
+
+1. Request or generate TLS certificates (in PEM format).
+
+   To generate self-signed certificates, we recommend that you **use our [scripts for generating TLS certificates](https://github.com/SkylineCommunications/generate-tls-certificates)**, available on GitHub. There is a version of the script for Linux and for Windows machines. The script requires two tools: *openssl* and the *Java keytool*. Both of these can run on Linux and Windows.
+
+1. Follow the steps below **for each of the NATS nodes** in the cluster to enable TLS encryption:
+
+   1. Update the **nats-server.config** file to incorporate the TLS section within the cluster segment.
+
+      This adjustment is necessary to accommodate the specified location of the PEM files. For example:
+
+      ```
+      cluster {
+        tls {
+          cert_file: "absolute\\path\\to\\certificate_file\\certificate.pem"
+          key_file:  "absolute\\path\\to\\private-key_file\\private-key.pem"
+          ca_file:   "absolute\\path\\to\\rootCA_file\\rootCA.pem"
+        }
+      }
+      ```
+
+      > [!NOTE]
+      > The `ca_file` property is only required when self-signed certificates are used.
+
+   1. In the *Services* tab in the Windows Task Manager, stop the NATS service.
+
+   1. In the *Services* tab in the Windows Task Manager, stop the NAS services.
+
+   1. In the *Services* tab in the Windows Task Manager, start the NATS service.
+
+      This should automatically trigger the start of the NAS service as well.
+
+   1. Verify whether TLS encryption is enabled for inter-node communication:
+
+      1. Go to `http://localhost:8222/varz` in a browser.
+
+      1. Check whether the `urls` property encompasses all the IP addresses of the NATS nodes configured with TLS.
+
+      1. Check whether the `tls_required` and `tls_verify` properties are appropriately configured to **true**.
+
+      For example:
+
+      ```json
+        "cluster": {
+          "urls": [
+            "[NATS_NODE_IP_ADDRESS]:6222",
+          ],
+          "tls_required": true,
+          "tls_verify": true
+        }
+      ```
+
+## Enabling DataMiner-to-NATS node TLS communication
+
+> [!IMPORTANT]
+> TLS encryption for communication between DataMiner and NATS nodes is supported from DataMiner 10.4.3 [CU0]/10.5.0 onwards<!-- RN 38302 -->, with CA-signed certificates (Certificate Authority-Signed Certificates). Avoid using self-signed certificates for this, as this may result in DataMiner startup failures.
```

Comment:
I would rephrase 'Avoid using self-signed certificates for this, as this may result in DataMiner startup failures.' to something along the line of 'For DataMiner 10.4.3, only CA-signed certificates are supported. Using a self-signed certificate will cause DataMiner to fail to start up'.

This specific issue with self-signed certs should be fixed from 10.4.4 onwards.

Reply:
I've rephrased the note so we don't promise an exact version with the fix, as I haven't seen a 10.4.4 RN for this yet. As soon as there is an RN for this, we can update this.